pipert/utils/logger_utils.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+from logging.handlers import SocketHandler, TimedRotatingFileHandler
+
+
+def create_parent_logger(name):
+    logger_type = os.environ.get('LOGGER', 'file').lower()
+    return getattr(LoggerGenerator, f"_create_{logger_type}_logger")(name)
+
+
+class LoggerGenerator:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def _create_cutelog_logger(name):
+        logger = logging.getLogger(name)
+        logger.setLevel(logging.DEBUG)
+        logger.propagate = False
+        socket_handler = SocketHandler('127.0.0.1', 19996)
+        socket_handler.setFormatter(logging.Formatter(
+            "%(asctime)s - %(levelname)s - %(name)s - %(message)s"))
+        logger.addHandler(socket_handler)
+        return logger
+
+    @staticmethod
+    def _create_file_logger(name):
+        class FileLogger(logging.Logger):
+            def __init__(self, name, level=logging.NOTSET):
+                super().__init__(name=name, level=level)
+
+            def getChild(self, suffix: str) -> logging.Logger:
+                return self.create_file_logger('-'.join((self.name, suffix)))
+
+            @staticmethod
+            def create_file_logger(name):
```

Comment:
Why is this a static method and what is the point of having two functions here (`getChild`, `create_file_logger`)?

Reply:
It is a bit weird but i will try to explain.

I made the class `LoggerGenerator` to contain all of the functions that create a new log type.
There is no point to make an instance of this class and it was easier to get a function by its name from a class object instead of getting all of the functions dynamically from a file and selecting the specific one.